myapp/src/pages/Discussion/CreateBoard.js

```diff
@@ -0,0 +1,150 @@
+import React, {useState} from "react";
+import axios from "axios";
+//import {Redirect} from "react-router-dom";
+import Padding from "../../components/Padding";
+import {Form as BootStrapForm, Col} from "react-bootstrap";
+import {NewCard} from "./BoardElements.js";
+
+function CreateBoard(props) {
+  const [board, setBoard] = useState(
+    {
+      groupName: "",
+      name: "",
+      description: ""
+    }
+  );
+
+  function handleChange(event) {
+    const {name, value} = event.target;
+    if (name == "group") {
+      setBoard(
+        {
+          groupName: value,
+          name: board.name,
+          description: board.description
+        }
+      );
+    }
+    if (name == "name") {
+      setBoard(
+        {
+          groupName: board.groupName,
+          name: value,
+          description: board.description
+        }
+      );
+    }
+    if (name == "description") {
+      setBoard(
+        {
+          groupName: board.groupName,
+          name: board.name,
+          description: value
+        }
+      );
+    }
+  }
+
+  async function makePostCall(thread) {
+    try {
+      const response = await axios.post('http://localhost:5000/discussion', thread);
+      return response;
+    }
+    catch (error) {
+      console.log(error);
+      return false;
+    }
+  }
+
+  function postThread(thread) {
+    if (thread.groupName == "") {
+      alert("Please enter a group name");
+      return;
+    }
+    if (thread.name == "") {
+      alert("Please enter a thread name");
+      return;
+    }
+    if (thread.description == "") {
+      alert("Please enter a description");
+      return;
+    }
+    makePostCall(thread).then(result => {
+      if (result) {
+        console.log(result);
+        //redirect
+      }
+    });
+  }
+
+
+  function submitForm() {
```

Comment:
The same thing as above regarding creating an array

Reply:
You could look at what I've got for the add user form for team roster regarding an array and mapping through!